File: datasets/rb/HumanEval_16_count_distinct_characters.rb
# Given a string, find out how many distinct characters (regardless of case) does it consist of
# >>> count_distinct_characters('xyzXYZ')
# 3
# >>> count_distinct_characters('Jerry')
# 4
def count_distinct_characters(string)
  string.downcase.chars.uniq.length
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_count_distinct_characters
    candidate = method(:count_distinct_characters)
    assert_equal(0, candidate.call(''))
    assert_equal(5, candidate.call('abcde'))
    assert_equal(5, candidate.call('abcdecadeCADE'))
    assert_equal(1, candidate.call('aaaaAAAAaaaa'))
    assert_equal(5, candidate.call('Jerry jERRY JeRRRY'))
  end
end

File: datasets/rs/HumanEval_16_count_distinct_characters.rs
/// Given a string, find out how many distinct characters (regardless of case) does it consist of
/// >>> count_distinct_characters('xyzXYZ')
/// 3
/// >>> count_distinct_characters('Jerry')
/// 4
fn count_distinct_characters(string: String) -> isize {
    let mut distinct_characters = HashSet::new();
    for c in string.chars() {
        distinct_characters.insert(c.to_lowercase().to_string());
    }
    distinct_characters.len() as isize

}

fn main() {
    let candidate = count_distinct_characters;
    assert_eq!(candidate(String::from("")), 0);
    assert_eq!(candidate(String::from("abcde")), 5);
    assert_eq!(candidate(String::from("abcdecadeCADE")), 5);
    assert_eq!(candidate(String::from("aaaaAAAAaaaa")), 1);
    assert_eq!(candidate(String::from("Jerry jERRY JeRRRY")), 5);
}
